File: APP_EFORM/app/src/main/java/com/example/eform_mobile/api/ApiService.java
package com.example.eform_mobile.api;

import com.example.eform_mobile.models.AuthResponse;
import com.example.eform_mobile.models.LoginRequest;
import com.example.eform_mobile.models.Product;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface ApiService {

    @POST("auth/login")
    Call<AuthResponse> login(@Body LoginRequest loginRequest);

    @GET("products")
    Call<List<Product>> getProducts();

    @POST("products")
    Call<Product> createProduct(@Body Product product);
}

File: APP_EFORM/lib/services/api_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../models/product.dart';

class ApiService {
  static const String baseUrl = 'http://localhost:8080';

  // LOGIN
  Future<bool> login(String email, String password) async {
    final response = await http.post(
      Uri.parse('$baseUrl/auth/login'),
      headers: {
        'Content-Type': 'application/json',
      },
      body: jsonEncode({
        'email': email,
        'password': password,
      }),
    );

    if (response.statusCode == 200) {
      return true;
    }

    return false;
  }

  // LISTAR PRODUCTOS
  Future<List<Product>> getProducts() async {
    final response = await http.get(
      Uri.parse('$baseUrl/products'),
    );

    if (response.statusCode == 200) {
      List<dynamic> data = jsonDecode(response.body);

      return data
          .map((item) => Product.fromJson(item))
          .toList();
    }

    throw Exception('Error cargando productos');
  }

  // CREAR PRODUCTO
  Future<bool> createProduct({
    required String nombre,
    required String descripcion,
    required double precio,
    required int stock,
  }) async {
    final response = await http.post(
      Uri.parse('$baseUrl/products'),
      headers: {
        'Content-Type': 'application/json',
      },
      body: jsonEncode({
        'nombre': nombre,
        'descripcion': descripcion,
        'tipoTela': '',
        'imageUrl': '',
        'tallasDisponibles': ['M'],
        'precio': precio,
        'stock': stock,
      }),
    );

    return response.statusCode == 200 ||
        response.statusCode == 201;
  }
}